analyzers/src/SonarAnalyzer.CSharp/Rules/SwaggerActionReturnType.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            if (!compilationStart.Compilation.Assembly.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_ApiConventionTypeAttribute)
+                && compilationStart.Compilation.ReferencesAll(KnownAssembly.MicrosoftAspNetCoreMvcCore, KnownAssembly.SwashbuckleAspNetCoreSwagger))
+            {
+                compilationStart.RegisterSymbolStartAction(symbolStart =>
+                {
+                    if (IsControllerCandidate(symbolStart.Symbol))
+                    {
+                        symbolStart.RegisterSyntaxNodeAction(nodeContext =>
+                        {
+                            var methodDeclaration = (MethodDeclarationSyntax)nodeContext.Node;
+                            if (InvalidMethod(methodDeclaration, nodeContext) is { } method)
+                            {
+                                nodeContext.ReportIssue(Rule, methodDeclaration.Identifier, additionalLocations: method.ResponseInvocations, GetMessage(method.Symbol));
+                            }
+                        }, SyntaxKind.MethodDeclaration);
+                    }
+                }, SymbolKind.NamedType);
+            }
+        });
+
+    private static InvalidMethodResult InvalidMethod(BaseMethodDeclarationSyntax methodDeclaration, SonarSyntaxNodeReportingContext nodeContext)
+    {
+        var responseInvocations = Builds200Responses(methodDeclaration, nodeContext.SemanticModel);
+        return responseInvocations.Length == 0
+               || nodeContext.SemanticModel.GetDeclaredSymbol(methodDeclaration, nodeContext.Cancel) is not { } method
+               || !method.IsControllerMethod()
+               || !method.ReturnType.DerivesOrImplementsAny(ControllerActionReturnTypes)
+               || method.GetAttributesWithInherited().Any(x => x.AttributeClass.DerivesFrom(KnownType.Microsoft_AspNetCore_Mvc_ApiConventionMethodAttribute)
+                                                               || HasApiExplorerSettingsWithIgnoreApiTrue(x)
+                                                               || HasProducesResponseTypeAttributeWithReturnType(x)
+                                                               || HasSwaggerResponseAttributeWithReturnType(x))
+                   ? null
+                   : new InvalidMethodResult(method, responseInvocations);
+    }
+
+    private static SyntaxNode[] Builds200Responses(SyntaxNode node, SemanticModel model)
+    {
+        return ActionResultInvocations().Concat(ObjectCreationInvocations()).Concat(ResultMethodsInvocations()).ToArray();
+
+        IEnumerable<SyntaxNode> ActionResultInvocations() =>
+            node.DescendantNodes()
+                .OfType<InvocationExpressionSyntax>()
+                .Where(x => ActionResultMethods.Contains(x.GetName())
+                            && x.ArgumentList.Arguments.Count > 0
+                            && model.GetSymbolInfo(x.Expression).Symbol.IsInType(KnownType.Microsoft_AspNetCore_Mvc_ControllerBase));
```

Comment:
By this, do you mean to check if the method used to build the response receives a value parameter? Because I would rather rework the logic to look for the `value` parameter than hardcode all the accepted method descriptions.

Reply:
The documentation does not show these. You can find them in source code: https://github.com/dotnet/aspnetcore/blob/8333a5b3ed07e73eb9342d994988bd9527039f11/src/Mvc/Mvc.Core/src/ControllerBase.cs#L2149